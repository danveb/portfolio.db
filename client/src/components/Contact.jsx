import "../styles/Contact.scss"; 

const Contact = () => {  
    return (
        <div id="contact" className="contact">
            <div className="contact__wrapper">
                <div className="contact__main">
                    <div className="contact__title">
                        <h1>Send Me A Note<span>.</span></h1>
                    </div>
                    <div className="contact__intro">
                        <p>I'm open to discussing new opportunities to work together. Feel free to send a note and I'll do my best to get back to you!</p>
                        <a href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!" target="_blank" rel="noreferrer noopener" className="cta">Say Hello</a>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Contact