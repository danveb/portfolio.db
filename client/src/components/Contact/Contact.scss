.contact {
    height: calc(100vh - 50px); 
    display: flex; 
    flex-direction: column; 
    background-color: var(--alice-blue); 

    .container {
        height: 70vh; 
        margin: 30px 30px 0 30px; 
        background-color: var(--alice-blue); 

        .contact-wrapper {
            height: 100%; 
            padding: 40px 60px; 
            display: flex; 
            justify-content: center; 
            align-items: center; 
            flex-direction: column; 

            @media screen and (max-width: 480px) {
                padding: 20px; 
            }

            .contact-headline {

                h1 {
                    color: var(--raisin-black); 
                    font-size: 40px; 
                    margin: 10px 0;

                    span {
                        color: var(--bottle-green); 
                    }

                    @media screen and (max-width: 768px) {
                        font-size: 30px; 
                    }
                    
                    @media screen and (max-width: 480px) {
                        font-size: 20px; 
                    }
                }
            }

            .contact-introduction {
                max-width: 768px; 
                width: 50%; 

                @media screen and (max-width: 768px) {
                    width: 75%; 
                }
                
                @media screen and (max-width: 480px) {
                    width: 100%; 
                }

                p {
                    color: var(--cadet-blue-crayola); 
                    text-align: center; 
                    margin: 10px 0;
                    font-size: 18px; 

                    @media screen and (max-width: 768px) {
                        font-size: 16px; 
                    }
                    
                    @media screen and (max-width: 480px) {
                        font-size: 14px; 
                    }
                }
            }

            .cta {
                display: flex; 
                justify-content: center; 
                align-items: center; 
                flex-direction: row; 
                margin: 20px 0; 
                border: 1px solid var(--bottle-green); 
                border-radius: 55px; 
                padding: 15px 25px; 
                background-color: var(--raisin-black); 
                color: var(--alice-blue); 
                text-transform: uppercase; 
                cursor: pointer; 
                text-decoration: none; 
                
                @media screen and (max-width: 768px) {
                    margin: 15px 0; 
                    padding: 10px 20px; 
                    font-size: 14px; 
                }
                
                @media screen and (max-width: 480px) {
                    font-size: 12px; 
                    margin: 10px 0;
                }
            }
        }
    }
}