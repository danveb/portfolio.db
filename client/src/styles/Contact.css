.contact {
  width: 100%; 
  height: 100%; 
}

.contact__wrapper {
  max-width: 768px; 
  margin: 0 auto; 
}

.contact__main {
  height: 100%; 
  width: 100%; 
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  gap: 20px; 
  margin: 40px 0; 
  position: relative; 
}

.contact__title {
  font-family: "Mulish", sans-serif;
  font-size: 20px; 
  color: var(--alice-blue); 
  letter-spacing: 0.5px; 
}

.contact__intro {
  font-family: inherit; 
  font-size: 16px; 
  font-weight: 300; 
  color: #d4d4d4; 
  letter-spacing: 1px; 
  line-height: 1.5rem; 
}

.sidebar {
  background-color: rgba(35, 55, 75, 0.9);
  color: #fff;
  padding: 6px 10px;
  font-family: monospace;
  z-index: 2;
  position: absolute;
  margin: 12px;
  font-size: 10px; 
}

.map-container {
  height: 45vh; 
  width: 100%; 
  margin-bottom: 40px; 
}

@media screen and (max-width: 1024px) {
  .contact__wrapper {
    padding: 0 20px; 
  }
}

@media screen and (max-width: 768px) {
  .contact__title {
    font-size: 16px; 
  }

  .contact__intro {
    font-size: 14px; 
  }
}