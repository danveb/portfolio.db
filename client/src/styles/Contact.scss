.contact {
    background-color: var(--alice-blue); 

    .contact__wrapper {
        max-width: 1200px; 
        height: 100%; 
        margin: 30px auto; 

        .contact__main {
            display: flex; 
            flex-direction: column; 
            justify-content: center; 
            align-items: center; 
            max-width: 590px; 
            margin: auto; 

            .contact__title {
                h1 {
                    color: var(--raisin-black); 
                    font-size: 38px; 
                    margin: 10px 0; 

                    span {
                        color: var(--bottle-green); 
                    }
                }
            }

            .contact__intro {
                width: 75%; 
                color: var(--raisin-black); 
                display: flex; 
                flex-direction: column; 
                justify-content: center; 
                align-items: center; 

                p {
                    color: var(--cadet-blue-crayola); 
                    text-align: center; 
                    margin: 20px 0;
                    font-size: 18px; 

                    @media screen and (max-width: 768px) {
                        font-size: 16px; 
                    }
                    
                    @media screen and (max-width: 480px) {
                        font-size: 14px; 
                    }
                }

                .cta {
                    margin: 20px 0; 
                    border: 1px solid var(--bottle-green); 
                    border-radius: 55px; 
                    padding: 15px 25px; 
                    background-color: var(--raisin-black); 
                    color: var(--alice-blue); 
                    text-transform: uppercase; 
                    cursor: pointer; 
                    text-decoration: none; 
                    
                    @media screen and (max-width: 768px) {
                        margin: 15px 0; 
                        padding: 10px 20px; 
                        font-size: 14px; 
                    }
                    
                    @media screen and (max-width: 480px) {
                        font-size: 12px; 
                        margin: 10px 0;
                    }
                }
            }
        }
    }
}