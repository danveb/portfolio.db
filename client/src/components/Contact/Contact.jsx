import { socials } from "../../constants/socials"; 
import "./Contact.scss"; 

const Contact = () => {  
    return (
        <div id="contact" className="contact">
            <div className="container">
                <div className="contact-wrapper">
                    <div className="contact-headline">
                        <h1>Send Me A Note<span>.</span></h1>
                    </div>
                    <div className="contact-introduction">
                        <p>I'm open to discussing new opportunities to work together. Feel free to send a note and I'll do my best to get back to you!</p>
                    </div>
                    <a href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!" target="_blank" rel="noreferrer noopener" className="cta">Say Hello</a>
                </div>
            </div>
            <div className="footer">
                <div className="footer-wrapper">
                    <div className="footer-headline">
                        <h1>Social Links<span>.</span></h1>
                    </div>
                    <div className="footer-social">
                        <ul>
                            {socials.map((social) => (
                                <li key={social.id}>
                                    <a href={social.url} target={social.target} rel={social.rel}>{<social.icon />}</a>
                                </li>
                            ))}
                        </ul>
                    </div>
                    <div className="footer-mr">
                        <p>Portfolio design & development - Danny Bae</p>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Contact