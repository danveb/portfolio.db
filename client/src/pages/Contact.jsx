import React, { useRef, useState, useEffect } from "react"; 
import mapboxgl from "!mapbox-gl" // eslint-disable-line import/no-webpack-loader-syntax
import "../styles/Contact.css"; 
import { Button } from "../components";

mapboxgl.accessToken = process.env.REACT_APP_MAPBOX_API; 

export default function Contact() {
  // initialize mapbox state/ref
  const mapContainer = useRef(null); 
  const map = useRef(null); 
  const [lat, setLat] = useState(41.8781);
  const [lng, setLng] = useState(-87.6298); 
  const [zoom, setZoom] = useState(11); 

  // useEffect
  useEffect(() => {
    if(map.current) return; // initialize map just once
    map.current = new mapboxgl.Map({
      container: mapContainer.current,
      style: "mapbox://styles/mapbox/dark-v11", 
      center: [lng, lat], 
      zoom: zoom, 
    }); 

    if(!map.current) return; // wait for map to initialize 
    map.current.on("move", () => {
      setLng(map.current.getCenter().lng.toFixed(4));
      setLat(map.current.getCenter().lat.toFixed(4)); 
      setZoom(map.current.getZoom().toFixed(2)); 
    });
  }, [lat, lng, zoom]); 

  return (
    <div className="contact">
      <div className="contact__wrapper">
        <div className="contact__main">
          <div className="contact__title">
            <h1>send me a note<span className="dot--green">.</span> üìù</h1>
          </div>
          <div className="breakline"></div>
          <div className="contact__intro">
            <p>I'm in the Chicago area, but open to discussing new opportunities to work together regardless of location. Feel free to send a note and I'll do my best to get back to you. </p>
          </div>
          <Button title="let's email" url="mailto:danibae91@gmail.com?subject=üë®‚Äçüíª  hey Danny, let's work together!" />
        </div>
        <div className="sidebar">
        Lng: {lng} | Lat: {lat}
        </div>
        <div ref={mapContainer} className="map-container" />
      </div>
    </div>
  )
}