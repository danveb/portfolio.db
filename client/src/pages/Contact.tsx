import { MapboxGL, Note } from "../components";
import "../styles/Contact.css"; 

export default function Contact() {
  return (
    <div className="contact">
      <div className="contact__wrapper">
        <section className="contact__main">
          <div className="contact__title">
            <h1>send me a note<span className="dot--green">.</span> üìù</h1>
          </div>
          <div className="breakline"></div>
          <div className="contact__intro">
            <p>I'm in the Chicago area, but open to discussing new opportunities to work together regardless of location. Feel free to send a note and I'll do my best to get back to you. </p>

          </div>
        </section>
        <MapboxGL />
        <Note />
      </div>
    </div>
  )
}