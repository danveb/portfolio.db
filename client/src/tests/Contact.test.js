import { render } from "@testing-library/react"; 
import "@testing-library/jest-dom/extend-expect";
import Contact from "../components/Contact/Contact"; 

// smoke test
it("renders without crashing", () => {
    render(<Contact />); 
}); 

// snapshot test
test("matches snapshot", () => {
    const { asFragment } = render(<Contact />); 
    expect(asFragment()).toMatchSnapshot(); 
});

// <h1>Send Me A Note<span>.</span></h1>
test("should display the h1 tag", () => {
    const { getByText } = render(<Contact />);
    getByText("Send Me A Note"); 
}); 

// <a href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!" target="_blank" rel="noreferrer noopener" className="cta">Say Hello</a> 
test("should display the anchor tag", () => {
    const { getByText } = render(<Contact />); 
    const anchor = getByText("Say Hello"); 
    expect(anchor).toHaveClass("cta"); 
    expect(anchor).toBeInTheDocument(); 
});