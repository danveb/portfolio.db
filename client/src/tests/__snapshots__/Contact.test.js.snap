// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="contact"
    id="contact"
  >
    <div
      class="container"
    >
      <div
        class="contact-wrapper"
      >
        <div
          class="contact-headline"
        >
          <h1>
            Send Me A Note
            <span>
              .
            </span>
          </h1>
        </div>
        <div
          class="contact-introduction"
        >
          <p>
            I'm open to discussing new opportunities to work together. Feel free to send a note and I'll do my best to get back to you!
          </p>
        </div>
        <a
          class="cta"
          href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!"
          rel="noreferrer noopener"
          target="_blank"
        >
          Say Hello
        </a>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="footer-wrapper"
      >
        <div
          class="footer-headline"
        >
          <h1>
            Social Links
            <span>
              .
            </span>
          </h1>
        </div>
        <div
          class="footer-social"
        >
          <ul>
            <li>
              <a
                href="https://www.github.com/danveb"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="GitHubIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://www.linkedin.com/in/daniel-eze-bae"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LinkedInIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="mailto:danibae91@gmail.com"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AlternateEmailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10h5v-2h-5c-4.34 0-8-3.66-8-8s3.66-8 8-8 8 3.66 8 8v1.43c0 .79-.71 1.57-1.5 1.57s-1.5-.78-1.5-1.57V12c0-2.76-2.24-5-5-5s-5 2.24-5 5 2.24 5 5 5c1.38 0 2.64-.56 3.54-1.47.65.89 1.77 1.47 2.96 1.47 1.97 0 3.5-1.6 3.5-3.57V12c0-5.52-4.48-10-10-10zm0 13c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="footer-mr"
        >
          <p>
            Portfolio design & development - Danny Bae
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
